### DISCUSSION TITLE
[Plugin Request] Combine all queries from multiple source files into a single `.graphql` file

### DISCUSSION THREAD

**Discussion by bkniffler** [dotansimha/graphql-code-generator#3669] [General]
I'm using and loving the code generator. Since I'm using Hasura, I'm looking for a quick way to combine all queries from my tsx/ts files into a single .graphql file to upload to my Hasura allowed query configuration (https://hasura.io/docs/1.0/graphql/manual/deployment/allow-list.html#adding-or-removing-a-query-in-allow-list). 

Would love to see a simple plugin that allows for the combination of all graphql queries into a single .graphql file.

**Comment by emielvanseveren** [comment]
I would also love this feature!

**Comment by dotansimha** [comment]
So it's something like `scheme-ast` plugin, but for documents? @emielvanseveren @bkniffler

**Comment by emielvanseveren** [comment]
Yes, my usecase would be equal to the one of bkniffler. Hasura has a feature to restrict the allowed queries. Users shouldn't be able to create custom queries. Logically this would be all these used in the application. It would be nice if all these queries could be gathered automatically and outputted to a single file. Maybe that's not completely related to graphql-code-generator tho..
